lib/ffmpeg/encoderbase.py

```diff
@@ -86,6 +86,12 @@ def gen_output_opts(self):
       opts += " -look_ahead_depth {ladepth}"
     if vars(self).get("vforced_idr", None) is not None:
       opts += " -force_key_frames expr:1 -forced_idr 1"
+    if vars(self).get("ref_type", None) is not None:
+      opts += " -int_ref_type {ref_type}"
+    if vars(self).get("ref_type", None) is not None:
+      opts += " -int_ref_cycle_size {ref_size}"
+    if vars(self).get("ref_type", None) is not None:
+      opts += " -int_ref_cycle_dist {ref_dist}"
```

Comment:
copy/paste error... all these conditionals are exactly the same.

The intra refresh type, size, and dist depend on each other.  That is, it does not make sense to have one but not the others.  Therefore, how about a variable like `intref = dict(type = T, size = S, dist = D)`.  Then do:

```python
if vars(self).get("intref", None) is not None:
  opts += " -int_ref_type {intref[type]}"
  opts += " -int_ref_cycle_size {intref[size]}"
  opts += " -int_ref_cycle_dist {intref[dist]}"
```